### DISCUSSION TITLE
Tag Tracking with Wildcard and Semantic Versioing doesnt work as expected

### DISCUSSION THREAD

**Discussion by Spazzy757** [argoproj/argo-cd#20595] [General]
We have a case where we use different suffixes to designate different environments i.e

- `v*.*.*-stag`
- `v*.*.*-prod`

According to [SemVar](https://semver.org/) this suffix falls into the "Additional Metadata" category. 

However with ArgoCD when we designate a `targetRevision: "v*.*.*-stag"` It seems to pick up ALL tags with `v*.*.*` and completely ignores the suffix.

According to the docs it says:
> For Git, all versions are Git references but tags [Semantic Versions](https://semver.org/) can also be used

Should the above scenario work, or am I misunderstanding the wildcard implementation

**Comment by holgerson97** [comment]
Is there any revision without the suffix in the repository?

**Reply by Spazzy757** [reply]
None at all, all our tags/releases have a suffix

**Reply by holgerson97** [reply]
I think "v" Prefix is not supported, as stated in SemVer: 

[SemVer](https://semver.org/#is-v123-a-semantic-version)
> Is “v1.2.3” a semantic version?
> No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

**Reply by Spazzy757** [reply]
We get the same result without the `v` as we do with it, it just picks up the latest tag, regardless of if it matches fully

**Reply by holgerson97** [reply]
So you also tried with tags without Prefix v?

**Reply by Spazzy757** [reply]
Yup, and changed the targetRevision to be: `*.*.*-stag`

**Reply by Spazzy757** [reply]
also tried `0.*.*-stag` but prod still picks it up as well with `0.*.*-prod`

**Comment by jose-arcatar** [comment]
I got the same problem, hopefully this get added in a future release

**Comment by Spazzy757** [comment]
Seems there is an open issue about this: https://github.com/argoproj/argo-cd/issues/18899

**Comment by Spazzy757** [comment]
I created a PR: that should fix this: https://github.com/argoproj/argo-cd/pull/20862

**Reply by vinitakiddom** [reply]
Which Argo version has this worked in? When I try to use target rev with v* or < > anything other than explicit it does not like it. I tried argocd image updater but it only works with kustomize image tags.

```
Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = Unable to resolve '1.x.x' to a commit SHA
```
